packages/patternfly-4/react-core/src/components/Form/FormGroup.js

```diff
@@ -0,0 +1,168 @@
+        }
+      }
+    } else if (Array.isArray(props.children)) {
+      return childrenArrayContains(props.children, 'id', `At least one of Form Group children should have an id.`);
+    } else if (!props.children.props.id) {
+      return new Error(`A child element of Form Group has to have an id.`);
+    }
+    return null;
+  },
+  /** Additional classes added to the FormGroup. */
+  className: PropTypes.string,
+  /** Label text before the field. */
+  label: PropTypes.oneOfType([PropTypes.string, PropTypes.node, PropTypes.func]),
+  /** Sets the FormGroup required. */
+  isRequired: PropTypes.bool,
+  /** Sets the FormGroup isValid. */
+  isValid: PropTypes.bool,
+  /** Sets the FormGroup isInline. */
+  isInline: PropTypes.bool,
+  /** Helper text after the field. It can be a simple text, a function or an object. */
+  helperText: PropTypes.oneOfType([PropTypes.string, PropTypes.node, PropTypes.func]),
+  /** Helper text after the field when the field is isValid. It can be a simple text, a function or an object. */
+  helperTextInvalid: PropTypes.oneOfType([PropTypes.string, PropTypes.node, PropTypes.func]),
+  /** ID of the included field. It has to be the same for proper working. */
+  fieldId: props => {
+    if (props.label && typeof props.label === 'string' && !props.fieldId) {
+      return new Error('FormGroup requires fieldId if label is string.');
+    }
+    if (props.fieldId && typeof props.fieldId !== 'string') {
+      return new Error(`FormGroup fieldId must be string. Received ${props.fieldId} which is ${typeof props.fieldId}.`);
+    }
+    return null;
+  }
+};
+
+const defaultProps = {
+  children: null,
+  className: '',
+  label: undefined,
+  isRequired: false,
+  isValid: true,
+  isInline: false,
+  helperText: undefined,
+  helperTextInvalid: undefined,
+  fieldId: undefined
+};
+
+const renderHelperText = ({ helperText, classes, fieldId }) => {
+  if (typeof helperText === 'function') {
+    return helperText();
+  } else if (typeof helperText === 'object') {
+    return helperText;
+  }
+  return (
+    <p className={classes} id={`${fieldId}-helper`} aria-live="polite">
+      {helperText}
+    </p>
+  );
+};
+
+const renderLabel = ({ label, isRequired, fieldId }) => {
+  const classesRequired = css(styles.formLabelRequired);
+  const classesLabel = css(styles.formLabel);
+
+  if (typeof label === 'function') {
+    return label();
+  } else if (typeof label === 'object') {
+    return label;
+  }
+  if (isRequired) {
+    return (
+      <label className={classesLabel} htmlFor={fieldId}>
+        {label}
+        <span className={classesRequired} aria-hidden="true">
+          {ASTERISK}
+        </span>
+      </label>
+    );
+  }
+  return (
+    <label className={classesLabel} htmlFor={fieldId}>
+      {label}
+    </label>
+  );
+};
+
+const renderHorizontal = children => {
+  const classesHorizontal = css(styles.formHorizontalGroup);
+
+  return <div className={classesHorizontal}>{children}</div>;
```

Comment:
Optional: I prefer inline function calls over variable assigns (kinda better for memory work in JS)
```JSX
const renderHorizontal = children => (
  <div className={css(styles.formHorizontalGroup)}>{children}</div>
);
```